### PR TITLE
The purifying fire that will extinguish evil

### PR DESCRIPTION
**Pull request by ivanlegranbizarro**
his Pull Request aims to exorcise the project from the evil that possessed it.

### CODE REVIEW

**Review by nachoa88 — APPROVED** [review]

File: app/Http/Controllers/api/AuthController.php [hunk 1]
```diff
@@ -15,58 +15,53 @@
 
 class AuthController extends Controller
 {
-	private UserService $userService;
-	
-	public function __construct(UserService $userService)
-    {
-        $this->userService = $userService;
-    }
+  private UserService $userService;
```
Comment (jordimorillo) [inline app/Http/Controllers/api/AuthController.php:18]: 4 spaces is a tab (in php). In java are 2 spaces.

File: app/Http/Controllers/api/AuthController.php [hunk 2]
```diff
@@ -15,58 +15,53 @@
...
-			throw new HttpResponseException(response()->json([
-				'message' => __(
-					'Wrong User Identication or Password'
-				)], 401));
-		}
+  public function signin(SigninRequest $request): JsonResponse
+  {
 
-	}
+    if ($this->userService->checkUserCredentials($request->dni, $request->password) == True) {
```
Comment (jordimorillo) [inline app/Http/Controllers/api/AuthController.php:30]: true (in lowercase). is only "True" capitalized in python but not in php.

File: app/Http/Controllers/api/AuthController.php [hunk 3]
```diff
@@ -15,58 +15,53 @@
...
 
-    
-    
-    
+      $userID = $this->userService->getUserIDByDNI($request->dni);
+
+      if (empty($userID) == False) {
```
Comment (jordimorillo) [inline app/Http/Controllers/api/AuthController.php:34]: ===

File: app/Http/Controllers/api/AuthController.php [hunk 4]
```diff
@@ -15,58 +15,53 @@
...
+        $jwt = $this->userService->generateJWToken($userID);
+        $storedBool = $this->userService->storeUserIDAndTokenRedis($userID, $jwt);
+        if ($storedBool == True) {
```
Comment (jordimorillo) [inline app/Http/Controllers/api/AuthController.php:37]: ===

File: app/Http/Controllers/api/AuthController.php [hunk 5]
```diff
@@ -15,58 +15,53 @@
...
+          $userDTO = new UserDTO(
+            (string)$userID,
+            $jwt
+          );
+
+          return response()->json($userDTO, 200);
+        } else {
+          throw new HttpResponseException(response()->json([
```
Comment (jordimorillo) [inline app/Http/Controllers/api/AuthController.php:45]: Unify 401 exception in a single throw

**Review by jordimorillo — APPROVED** [review]
Very good refactor. Solve the details and merge it! :)

**Comment by ivanlegranbizarro** [comment]
I merge the PR without changing anything in the AuthController because Nacho and Mario are already solving this.

**Pull request merged**